lib/SILOptimizer/Mandatory/TFLowerGraph.cpp

```diff
@@ -802,10 +802,19 @@ static TF_Tensor *convertValuesToTensor(ArrayRef<APInt> elts,
                                    dtypeSize * totalElements);
 
   // Set up its contents, element-wise.
+  // FIXME: This will need a byte swap for big endian hosts.
   auto *ptr = (char *)TF_TensorData(tensor);
   for (auto elt : elts) {
-    // FIXME: This will need a byte swap for big endian hosts.
-    memcpy(ptr, elt.getRawData(), dtypeSize);
+    if (dtype == TF_BOOL) {
+      // Swift/LLVM uses a 1-bit APInt to represent a bool, but TF_BOOL is 1
+      // byte.
+      assert(sizeof(bool) == dtypeSize);
```

Comment:
Heh, if you want to be super pedantic about this, this code is incorrect (e.g. on PowerPC hosts using the IBM convention like Darwin/PPC).  The issue is that sizeof(bool) with the host and target may differ (PPC uses a 4 byte bool).

I'd suggest changing this to be something like:
if (dtype == TF_BOOL)
  elt = elt.zext(dtypeSize*8);

Reply:
This is a TF-level sanity check (I'm following the impl of `TF_DataTypeSize()` at https://github.com/tensorflow/tensorflow/blob/31e9d08398a2118537abc1a02f47376c4d2c7d9d/tensorflow/c/c_api.cc#L106, which in turn returns `sizeof<T>` at https://github.com/tensorflow/tensorflow/blob/31e9d08398a2118537abc1a02f47376c4d2c7d9d/tensorflow/core/framework/types.cc#L229, and believe `T` is `bool` for  TF_BOOL) -- not sure how "host vs target" plays a role here?

Thanks for the suggestion on using `zext`. Replaced the temp var `val` with it.

Reply:
With the new code based on `zext`, the assert on `sizeof(bool)` is no longer adding any value, so I removed it.

Regarding host vs target, I believe what you mean is: 
- say we are building for a TF backend to run on a machine where sizeof(bool) != 1. That's the target compiler setting
- I'm guess in that case we need to build swiftc on that same target setting, since swiftc calls into TF lib, and it need to agree with TF on sizeof(bool) in the TF runtime.